server/src/main/java/keywhiz/service/permissions/AlwaysAllowDelegatingPermissionCheck.java

```diff
@@ -32,14 +33,41 @@ public boolean isAllowed(Object source, String action, Object target) {
     return true;
   }
 
+  @Override
+  public boolean isAllowedForTargetType(Object source, String action, Class<?> targetType) {
+    boolean hasPermission = delegate.isAllowedForTargetType(source, action, targetType);
+
+    emitHistogramMetrics(hasPermission);
+
+    logger.info(
+        String.format("isAllowedByType Actor: %s, Action: %s, Target type: %s, Result: %s", source, action, targetType,
+            hasPermission));
```

Comment:
I can see a PermissionCheckContext class in our future, where you set all of the fields for which you have data available via a builder and the toString() belongs to the context class so that we don't have to manually print it out here.